crates/polars-expr/src/expressions/aggregation.rs

```diff
@@ -38,8 +49,103 @@ impl PhysicalExpr for AggregationExpr {
+            GroupByMethod::NanMax if !is_float => GroupByMethod::Max,
+            gb => gb,
+        };
+
+        match group_by {
+            GroupByMethod::Min => match s.is_sorted_flag() {
+                IsSorted::Ascending | IsSorted::Descending => {
+                    s.min_reduce().map(|sc| sc.into_series(s.name()))
+                },
+                IsSorted::Not => parallel_op_series(
+                    |s| s.min_reduce().map(|sc| sc.into_series(s.name())),
+                    s,
+                    allow_threading,
+                ),
+            },
+            #[cfg(feature = "propagate_nans")]
+            GroupByMethod::NanMin => parallel_op_series(
+                |s| {
+                    Ok(polars_ops::prelude::nan_propagating_aggregate::nan_min_s(
+                        &s,
+                        s.name(),
+                    ))
+                },
+                s,
+                allow_threading,
+            ),
+            #[cfg(not(feature = "propagate_nans"))]
+            GroupByMethod::NanMin => {
+                panic!("activate 'propagate_nans' feature")
+            },
+            GroupByMethod::Max => match s.is_sorted_flag() {
+                IsSorted::Ascending | IsSorted::Descending => {
+                    s.max_reduce().map(|sc| sc.into_series(s.name()))
+                },
+                IsSorted::Not => parallel_op_series(
+                    |s| s.max_reduce().map(|sc| sc.into_series(s.name())),
+                    s,
+                    allow_threading,
+                ),
+            },
+            #[cfg(feature = "propagate_nans")]
+            GroupByMethod::NanMax => parallel_op_series(
+                |s| {
+                    Ok(polars_ops::prelude::nan_propagating_aggregate::nan_max_s(
+                        &s,
+                        s.name(),
+                    ))
+                },
+                s,
+                allow_threading,
+            ),
+            #[cfg(not(feature = "propagate_nans"))]
+            GroupByMethod::NanMax => {
+                panic!("activate 'propagate_nans' feature")
+            },
+            GroupByMethod::Median => s.median_reduce().map(|sc| sc.into_series(s.name())),
+            GroupByMethod::Mean => Ok(s.mean_reduce().into_series(s.name())),
+            GroupByMethod::First => Ok(if s.is_empty() {
+                Series::full_null(s.name(), 1, s.dtype())
+            } else {
+                s.head(Some(1))
+            }),
+            GroupByMethod::Last => Ok(if s.is_empty() {
+                Series::full_null(s.name(), 1, s.dtype())
+            } else {
+                s.tail(Some(1))
+            }),
+            GroupByMethod::Sum => parallel_op_series(
+                |s| s.sum_reduce().map(|sc| sc.into_series(s.name())),
+                s,
+                allow_threading,
+            ),
+            GroupByMethod::Groups => todo!(),
```

Comment:
I believe this one is `unreachable`. This will be removed one day when we drop all eager support of group-by.